### PR TITLE
Readthedocs

### PR DESCRIPTION
**Pull request by NathanTP**
This adds sphinx documentation to SRK. Once this goes in, I can setup a readthedocs page for more longform documentation (to complement the reference material in the godoc).

### CODE REVIEW

File: docs/source/QuickStart.rst [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+maintain a fork of this project to provide additional features. Go ahead and
+clone this repo anywhere in your filesystem, we'll use our home directory in
+these instructions:
+
+::
+
+   $ cd ~
+   $ git clone git@github.com:NathanTP/open-lambda.git
+
+Next, follow the instructions in open-lambda's README to compile and install
```
Comment (jssmith) [inline docs/source/QuickStart.rst]: Maybe link to readme, especially to help a casual reader.

File: docs/source/QuickStart.rst [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+With OpenLambda built, we can now configure SRK to use it. Open up
+``configs/srk.yaml`` and modify the ``service.faas.openLambda`` section to look
+as follows:
+
+::
+
+   faas:
+      openLambda:
+         olcmd: "~/open-lambda/ol"
+         oldir: "~/open-lambda/default-ol"
```
Comment (jssmith) [inline docs/source/QuickStart.rst]: Might be nice to explain what oldir is for.

File: docs/source/QuickStart.rst [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+<https://docs.aws.amazon.com/lambda/latest/dg/getting-started.html>`_. For SRK
+to work, you will need to provide an ARN role and optional VPC. You can set
+these values in ``configs/srk.yaml`` in the ``service.faas.awsLambda`` section.
+For example:
+
+::
+
+   faas:
+      awsLambda:
+         role: "arn:aws:iam::123459789012:role/service-role/my-service-role-ae04d032"
```
Comment (jssmith) [inline docs/source/QuickStart.rst]: We should probably specify what privileges need to be associated with the role. Maybe provide CLI commands to create the necessary role.

File: docs/source/QuickStart.rst [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+         vpc: "vpc-123456789abcdef,sg-123456789abcdef"
```
Comment (jssmith) [inline docs/source/QuickStart.rst]: How does the security group need to be configured?

File: docs/source/QuickStart.rst [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+=================================
+Now that we have both AWS Lambda and OpenLambda configured, we can switch
+between them by changing the ``default-provider`` option in
+``configs/srk.yaml``. To start with, let's switch to use local resources only (e.g. OpenLambda):
+
+::
+
+   default-provider: local
+
+To switch to AWS, you can instead set it to ``default-provider: aws``.
```
Comment (jssmith) [inline docs/source/QuickStart.rst]: Wondering... can I switch the provider with a command-line argument or an environment variable?

File: docs/source/Configuration.rst [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+This section contains descriptions of individual services, organized by service
+category.
+
+faas
+----------------------
+Function-as-a-Service.
+
+openLambda
+^^^^^^^^^^^^^^^^^^^^^^
+OpenLambda (OL) is an open source FaaS service that is built on Docker. OL is
```
[the remark this answers is not in the document]
Comment (NathanTP) [reply docs/source/Configuration.rst]: They use "Apache-licensed serverless computing project, written in Go and based on Linux containers". I changed the line to say "based on Linux containers", but the way we use it, it's really Docker. The rest I think is close enough.

File: docs/source/QuickStart.rst (hunk 1 above)
Comment (NathanTP) [reply docs/source/QuickStart.rst]: done

File: docs/source/QuickStart.rst (hunk 2 above)
Comment (NathanTP) [reply docs/source/QuickStart.rst]: fixed.

File: docs/source/QuickStart.rst (hunk 3 above)
Comment (NathanTP) [reply docs/source/QuickStart.rst]: I added a bit more docs for this and a link to the appropriate AWS documentation, but I'd love if you could flesh this out a bit more.

File: docs/source/QuickStart.rst (hunk 4 above)
Comment (NathanTP) [reply docs/source/QuickStart.rst]: no idea. I added some more stuff here. The VPC can really just be NULL unless you need something special so I changed it to default to that. Again, anything you can add to the docs here would be great.

File: docs/source/QuickStart.rst (hunk 5 above)
Comment (NathanTP) [reply docs/source/QuickStart.rst]: no. But we could. It's a relatively easy change in the code. I'll do that eventually, it's a good idea.

**Pull request merged**